desc/objectives/_free_boundary.py

```diff
@@ -0,0 +1,945 @@
+"""Objectives for solving free boundary equilibria."""
+
+import warnings
+
+import numpy as np
+from scipy.constants import mu_0
+
+from desc.backend import jnp
+from desc.compute import compute as compute_fun
+from desc.compute import get_params, get_profiles, get_transforms
+from desc.grid import LinearGrid
+from desc.nestor import Nestor
+from desc.objectives.objective_funs import _Objective
+from desc.singularities import (
+    DFTInterpolator,
+    FFTInterpolator,
+    virtual_casing_biot_savart,
+)
+from desc.utils import Timer, errorif, warnif
+
+from .normalization import compute_scaling_factors
+
+
+class VacuumBoundaryError(_Objective):
+    """Target for free boundary conditions on LCFS for vacuum equilibrium.
+
+    Computes the residuals of the following:
+
+    𝐁ₒᵤₜ ⋅ 𝐧 = 0
+    𝐁ₒᵤₜ² - 𝐁ᵢₙ² = 0
```

Comment:
The actual compute function adds an area element weighting. We document that similar thing in the equations here for the force balance objectives, and I think it would be good to include here too. So that it's obvious what the units of the residuals are.